x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/TranslatorHandler.java

```diff
@@ -29,19 +30,23 @@ public final class TranslatorHandler {
 
     private TranslatorHandler() {}
 
-    public Query asQuery(Expression e) {
+    public Query asQuery(LucenePushdownPredicates predicates, Expression e) {
         if (e instanceof TranslationAware ta) {
-            Query query = ta.asQuery(this);
+            Query query = ta.asQuery(predicates, this);
             return ta instanceof TranslationAware.SingleValueTranslationAware sv ? wrapFunctionQuery(sv.singleValueField(), query) : query;
         }
 
         throw new QlIllegalArgumentException("Don't know how to translate {} {}", e.nodeName(), e);
     }
 
     private static Query wrapFunctionQuery(Expression field, Query query) {
+        if (query instanceof SingleValueQuery) {
+            // Already wrapped
+            return query;
+        }
```

Comment:
I'm not super proud of this. I kind of thing we should remove this and have folks wrap the query they build themselves. But not in this PR.

Reply:
This exists so `Equals` can have a different behavior - it checks the value count of the synthetic source delegate.....


Wait. What if we remove one? Oh no.

Reply:
Ok. I've added a fix for this. I'll push a javadoc explaining it.

Reply:
I pushed 600257acc0c.